package.json

```diff
@@ -31,14 +31,19 @@
     },
     "./request-handler": "./dist/request-handler.js",
     "./video-types/*": "./video-types/*.d.ts",
+    "./dist/cjs/*": "./dist/cjs/*",
     "./dist/*": "./dist/*"
   },
   "scripts": {
     "clean": "rm -rf dist",
-    "watch": "npm run build:esm -- -w & npm run build:cjs -- -w",
-    "build": "npm run clean && (npm run build:esm && npm run build:cjs)",
-    "build:esm": "tsc --module es2020",
-    "build:cjs": "tsc src/main.ts src/webpack-loader.ts --module commonjs --outDir dist/cjs --strict  --target es2019 --esModuleInterop --declaration",
+    "watch": "npm run types:esm -- -w & npm run types:cjs -- -w & npm run build:esm -- --watch=forever & npm run build:cjs -- --watch=forever",
+    "types": "npm run types:esm & npm run types:cjs",
+    "types:esm": "tsc --outDir dist",
+    "types:cjs": "tsc --outDir dist/cjs",
+    "prebuild": "npm run clean && npm run types",
+    "build": "npm run build:esm && npm run build:cjs",
+    "build:esm": "esbuild src/*.ts* src/*/*.ts* src/*/*/*.ts* --outdir=dist --format=esm --target=es2020",
+    "build:cjs": "esbuild src/*.ts* src/*/*.ts* src/*/*/*.ts* --outdir=dist/cjs --platform=node --format=cjs --target=es2020 --define:import.meta.url=\\\"\\\"",
```

Comment:
starting using esbuild because we need to replace import.meta.url. if this is used in a CJS module Node throws an error